### PR TITLE
Simplify the jsg::MemoryTracker def/impl

### PR DESCRIPTION
**Pull request by jasnell**
Reduces the size of the inlined headers, moving impl details in the c++ file. Refines a few of the impl details.

### CODE REVIEW

**Review by hoodmane — APPROVED** [review]
Generally looks reasonable to me. I only read about half of the code that has been moved to look for changes, but it wasn't obvious to me that much has been both changed and moved other than the systematic return type to `void` changes.

**Comment by hoodmane** [comment]
Ah but this is a draft anyways.

File: src/workerd/jsg/memory.h [hunk 1]
```diff
@@ -257,490 +252,294 @@ class MemoryTracker final {
...
   KJ_DISALLOW_AS_COROUTINE_PARAM;
 
-  inline explicit MemoryTracker(v8::Isolate* isolate,
-                                v8::EmbedderGraph* graph)
-    : isolate_(isolate),
-      graph_(graph) {}
+  explicit MemoryTracker(v8::Isolate* isolate, v8::EmbedderGraph* graph);
 
-  inline kj::Maybe<MemoryRetainerNode&> getCurrentNode() const;
+  kj::Maybe<MemoryRetainerNode&> getCurrentNode() const KJ_LIFETIMEBOUND;
```
[the remark this answers is not in the document]
Comment (jasnell) [reply src/workerd/jsg/memory.h]: it's an alias for `[[clang::lifetimebound]]` (https://clang.llvm.org/docs/AttributeReference.html#lifetimebound) ... essentially the compiler will give us a warning if we try to use the dangling reference here beyond the lifetime of the `MemoryTracker` object.

**Comment by jasnell** [comment]
Folding these changes into one of the other PRs

**Pull request closed without merging**